src/index.ts

```diff
@@ -32,11 +32,11 @@ const _hasOwn = Object.prototype.hasOwnProperty;
  * @template T
  */
 export function map<T>(opt_initial: T | undefined) {
-  const obj = Object.create(null);
+  const object = Object.create(null);
   if (opt_initial) {
-    Object.assign(obj, opt_initial);
+    Object.assign(object, opt_initial);
   }
-  return obj;
+  return object;
```

Comment:
```suggestion
  return { ...opt_initial };
```

Reply:
That might work; need to double-check, but just so that we're on the same page, this is what we are referred to when we say that it returns a "map-like object":

https://github.com/google/closure-compiler/wiki/A-word-about-the-type-Object#map-like-objects